pkg/operator/configobservation/serviceca/observeserviceca.go

```diff
@@ -0,0 +1,61 @@
+package serviceca
+
+import (
+	"k8s.io/apimachinery/pkg/api/equality"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	"github.com/openshift/cluster-kube-controller-manager-operator/pkg/operator/configobservation"
+	"github.com/openshift/cluster-kube-controller-manager-operator/pkg/operator/operatorclient"
+	"github.com/openshift/library-go/pkg/operator/configobserver"
+	"github.com/openshift/library-go/pkg/operator/events"
+)
+
+const (
+	serviceCAFilePath      = "/etc/kubernetes/static-pod-resources/configmaps/service-ca/service-ca.crt"
+	serviceCAConfigMapName = "service-ca"
+)
+
+// ObserveServiceCA fills in serviceServingCert.CertFile with the path for a configMap containing the service-ca.crt
+func ObserveServiceCA(genericListers configobserver.Listers, recorder events.Recorder, existingConfig map[string]interface{}) (map[string]interface{}, []error) {
+	listers := genericListers.(configobservation.Listers)
+	errs := []error{}
+	prevObservedConfig := map[string]interface{}{}
+
+	topLevelServiceCAFilePath := []string{"serviceServingCert", "certFile"}
+	currentServiceCAFilePath, _, err := unstructured.NestedString(existingConfig, topLevelServiceCAFilePath...)
+	if err != nil {
+		errs = append(errs, err)
+	}
+	if len(currentServiceCAFilePath) > 0 {
+		if err := unstructured.SetNestedField(prevObservedConfig, currentServiceCAFilePath, topLevelServiceCAFilePath...); err != nil {
+			errs = append(errs, err)
+		}
+	}
+
+	observedConfig := map[string]interface{}{}
+
+	// add service-ca configmap mount path to ServiceServingCert if configmap exists
+	ca, err := listers.ConfigMapLister.ConfigMaps(operatorclient.TargetNamespace).Get(serviceCAConfigMapName)
+	if errors.IsNotFound(err) {
+		// do nothing because we aren't going to add a path to a missing configmap
+		return observedConfig, errs
+	}
+	if err != nil {
+		// we had an error, return what we had before and exit. this really shouldn't happen
+		return prevObservedConfig, append(errs, err)
+	}
+	if len(ca.Data) <= 0 {
```

Comment:
I think we want to make sure the expected key has data.

Reply:
4957ab4f I'm unsure of the behavior on the kube-controller-manager side if the key is set to the empty string.